tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Holders/RepoLabelHolder.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Sdk.Tools.CodeownersUtils.Constants;
+using Azure.Sdk.Tools.CodeownersUtils.Utils;
+
+namespace Azure.Sdk.Tools.CodeownersUtils.Holders
+{
+    /// <summary>
+    /// Holder for the repo label dictionary. Lazy init, it doesn't pull the data from the URI until first use.
+    /// The GetRepoLabelData is interesting because the repo/label data is a dictionary of hashsets but, both
+    /// the dictionary and all of the hashsets need to be case insensitive because GitHub is case insensitive
+    /// but case preserving when it comes to labels. This means that MyLabel, mylabel and mYlAbEl are effectively
+    /// the same to GitHub and, let's face it, the CODEOWNERS files are all over the place with exact vs non-exact
+    /// casing.
+    /// </summary>
+    public class RepoLabelHolder
```

Comment:
nit: Is Holder the right name for these classes? Would Cache be a more appropriate name?

Reply:
Cache probably would be. I can rename them, and their folder, easily enough and will do so.

Reply:
All of these changed from Holder to Cache as part of the 6th commit.